libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -24,14 +31,23 @@ export class BookSearchComponent implements OnInit {
 
   constructor(
     private readonly store: Store,
-    private readonly fb: FormBuilder
+    private readonly fb: FormBuilder,
+    private readonly cdr: ChangeDetectorRef,
   ) {}
 
   get searchTerm(): string {
     return this.searchForm.value.term;
   }
 
   ngOnInit(): void {
+
+    this.searchForm.valueChanges.pipe(map(val => val.term),debounceTime(500), distinctUntilChanged(),
+    takeUntil(this.destroy)).subscribe(value => {
+      this.instantSearchText = value;
```

Comment:
Are you using instantSearchText variable anywhere? Please remove, if not required.